### PR TITLE
Refactor directory structure

### PR DESCRIPTION
**Pull request by lscheinkman**
@brev I've finished refactoring `unicorn` directory structure as we discussed a few weeks ago. Basically the refactor will replace the words  `backend`, `frontend` and `lib` with `py`, `js` and `main`. See `README` for more details.
Here is as summary:
```
py/                 # Unicorn ModelRunner and support Python/C++ code here!
js/                 # Frontend+GUI that exposes NuPIC HTM functionality to the User
  browser/          # Javascript, HTML, CSS act as GUI inside electron's "Renderer Process"
  main/             # JS from electron's "Main Process" (i.e. engine services)
``` 
Please let me know if you like to change anything else.

### CODE REVIEW

File: unicorn/package.json [hunk 1]
```diff
@@ -105,27 +105,27 @@
...
     "babel-plugin-typecheck": "1.3.0",
-    "css-loader": "0.20.2",
-    "electron-debug": "0.2.1",
-    "electron-packager": "5.1.0",
+    "css-loader": "0.23.0",
+    "electron-debug": "0.5.0",
+    "electron-packager": "5.1.1",
     "electron-prebuilt": "0.33.3",
-    "esdoc": "0.4.3",
+    "esdoc": "1.0.0",
```
Comment (brev) [inline unicorn/package.json]: `npm run docs` doesn't work for me, I don't see `esdoc@1.0.0` on npmjs.org or in a github release. You could use a specific github URL?  Docs do successfully build for me with `esdoc@0.4.3`, but may be missing something you were looking for.

Comment (lscheinkman) [inline unicorn/package.json]: oops, not sure it got updated, I got the versions by using `npm update --save` and. 
I will fix that.

**Comment by brev** [comment]
looks good to me from JS/frontend point of view. My backend/PY is down right now, so I can't confirm build/test yet. Merge if you are confident, I will get there ASAP. thanks

**Comment by brev** [comment]
@lscheinkman :+1: Verified on my system for everything but cx-frozen-python.

**Pull request merged**